capsules/src/segger_rtt.rs

```diff
@@ -0,0 +1,194 @@
+//! Segger RTT implementation.
+//!
+//! RTT is a protocol for sending debugging messages to a connected host. The
+//! embedded platform configures a portion of memory in a special way, and then
+//! the host uses a JTAG connection to read the messages out of the chip's
+//! memory.
+//!
+//!	Receiving RTT Messages
+//!	----------------------
+//!
+//!	With the jlink tools, reciving RTT messages is a two step process. First,
+//!	open a JTAG connection with a command like:
+//!
+//!	    $ JLinkExe -device nrf52 -if swd -speed 1000 -autoconnect 1
+//!
+//!	Then, use the `JLinkRTTClient` tool in a different terminal to print the
+//!	messages:
+//!
+//!	    $ JLinkRTTClient
+//!
+//! Notes
+//! -----
+//!
+//! This capsule requires a timer, but the timer is only there to defer the
+//! `transmit_complete` callback until the next scheduler loop. In the future,
+//! if there is support for software interrupts or deferred calls in capsules,
+//! this timer should be removed.
+//!
+//! Usage
+//! -----
+//!
+//! ```
+//! let virtual_alarm_rtt = static_init!(
+//!     capsules::virtual_alarm::VirtualMuxAlarm<'static, nrf5x::rtc::Rtc>,
+//!     capsules::virtual_alarm::VirtualMuxAlarm::new(mux_alarm)
+//! );
+//!
+//! let rtt_memory = static_init!(
+//!     capsules::segger_rtt::SeggerRttMemory,
+//!     capsules::segger_rtt::SeggerRttMemory::new(b"Terminal\0",
+//!         &mut capsules::segger_rtt::UP_BUFFER,
+//!         b"Terminal\0",
+//!         &mut capsules::segger_rtt::DOWN_BUFFER)
+//! );
+//!
+//! let rtt = static_init!(
+//!     capsules::segger_rtt::SeggerRtt<VirtualMuxAlarm<'static, nrf5x::rtc::Rtc>>,
+//!     capsules::segger_rtt::SeggerRtt::new(virtual_alarm_rtt, rtt_memory,
+//!         &mut capsules::segger_rtt::UP_BUFFER,
+//!         &mut capsules::segger_rtt::DOWN_BUFFER)
+//! );
+//! virtual_alarm_rtt.set_client(rtt);
+//! ```
+
+use kernel::common::cells::{OptionalCell, TakeCell};
+use kernel::hil;
+use kernel::hil::time::Frequency;
+
+/// Buffer for transmitting to the host.
+pub static mut UP_BUFFER: [u8; 1024] = [0; 1024];
+
+/// Buffer for receiving messages from the host.
+pub static mut DOWN_BUFFER: [u8; 32] = [0; 32];
```

Comment:
I think `TX_BUFFER` and `RX_BUFFER` is better here

Reply:
Up and down matches the segger documentation and example code.

Reply:
Ok, then I see